modin/pandas/test/test_io.py

```diff
@@ -524,6 +552,66 @@ def test_read_csv_delimiters(
             **kwargs,
         )
 
+    # Datetime Handling tests
+    @pytest.mark.parametrize(
+        "parse_dates",
+        [
+            True,
+            False,
+            ["col2"],
+            ["col2", "col4"],
+            [1, 3],
+            pytest.param(
+                {"foo": ["col2", "col4"]},
+                marks=pytest.mark.xfail(
+                    Engine.get() != "Python",
+                    reason="Exception: Internal Error - issue #2073",
+                ),
+            ),
+        ],
+    )
+    @pytest.mark.parametrize("infer_datetime_format", [True, False])
+    @pytest.mark.parametrize("keep_date_col", [True, False])
+    @pytest.mark.parametrize(
+        "date_parser", [None, lambda x: pd.datetime.strptime(x, "%Y-%m-%d")]
+    )
+    @pytest.mark.parametrize("dayfirst", [True, False])
+    @pytest.mark.parametrize("cache_dates", [True, False])
+    def test_read_csv_datetime(
+        self,
+        request,
+        parse_dates,
+        infer_datetime_format,
+        keep_date_col,
+        date_parser,
+        dayfirst,
+        cache_dates,
+    ):
+        if request.config.getoption("--simulate-cloud").lower() != "off":
+            pytest.xfail("The reason of tests fail in `cloud` mode is unknown for now")
+        case_with_TypeError = isinstance(parse_dates, dict) and callable(date_parser)
+        case_with_TypeError_exc = list(io_ops_bad_exc)
+        case_with_TypeError_exc.remove(TypeError)
+
+        kwargs = {
+            "parse_dates": parse_dates,
+            "infer_datetime_format": infer_datetime_format,
+            "keep_date_col": keep_date_col,
+            "date_parser": date_parser,
+            "dayfirst": dayfirst,
+            "cache_dates": cache_dates,
+        }
+
+        eval_io(
+            filepath_or_buffer=pytest.csvs_names["test_read_csv_regular"],
+            fn_name="read_csv",
+            check_kwargs_callable=not callable(date_parser),
```

Comment:
Why we need in `check_kwargs_callable`?

Reply:
This is done because in this tests `date_parser` parameter can be passed as lambda function and it should be forwarded into `read_csv` function, not executed.